builder/internals.go

```diff
@@ -621,7 +621,7 @@ func (b *builder) create() (*daemon.Container, error) {
 	config := *b.Config
 
 	// Create the container
-	c, warnings, err := b.Daemon.Create(b.Config, hostConfig, "")
+	c, warnings, err := b.Daemon.Create(b.Config, hostConfig, "", true)
```

Comment:
Set `true` always adapt the `cpushares` for build, so if pass a value for cpu-shares larger than the maximum allowed value or smaller than the allowed value, it will adapt the `cpushares`  the to the allowed value.